app/javascript/components/SubmitButton.vue

```diff
@@ -1,7 +1,7 @@
 <template>
   <button
     @click="$emit('do-click')"
-    :class="[buttonClass,colormod,hovermod,widemod]"
+    :class="[buttonClass,colormod,hovermod,widthmod]"
```

Comment:
Creo que esto quedaría mas simple [ocupando style bindings de vue](https://vuejs.org/v2/guide/class-and-style.html) y escribiendo las clases de tailwind directamente. Podrías lograr lo mismo haciendo algo así (puede que no este 100% correcto pero debiese ser algo parecido):

```js
// Las clases que no cambian puedes ponerlas sin binding (:), porque no ocupan los props
class='text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline'
// y despues pones las que si necesitan. Aqui puedes acceder a los props para elegir entre clases
:class="[
    width === 'full' ? 'w-full' : 'w-normal',
    color === 'secondary' ? 'bg-indigo-500 hover:bg-indigo-700' : 'bg-blue-500 hover:bg-blue-700'
]"
```

El [operador ternario](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) funciona asi:

```js
X ? (retorna esto si X es true) : (retorna esto si X es false)
```
Entonces para cada linea de la lista en `:class` se evalúa el valor y se elige que retornar, después todas las clases retornadas de esa lista se agregan al componente. De esta forma te evitas todos los ifs y guardar los valores predeterminados en `data`.

Reply:
Me gusta esa forma de hacerlo, es más limpio

Reply:
Súper, no sabía que se podía hacer de esta forma, queda mucho mejor. Lo corregí, gracias!